tests/qchem/of_tests/test_convert.py

```diff
@@ -813,3 +813,321 @@ def dummy_ansatz(phis, wires):
+            np.array([1, 1, 1, 1, 1, -1, -1, -1, -1, -1, 1, 1, 1, 1, 1]),
+        ),
+        (
+            3,
+            6,
+            2,
+            np.array([28, 44, 52, 26, 42, 50, 25, 41, 49]),
+            np.array([1, 1, 1, -1, -1, -1, 1, 1, 1]),
+        ),
+    ],
+)
+def test_excited_configurations(electrons, orbitals, excitation, states_ref, signs_ref):
+    r"""Test if the _excited_configurations function returns correct states and signs."""
+    states, signs = qchem.convert._excited_configurations(electrons, orbitals, excitation)
+    assert np.allclose(states, states_ref)
+    assert np.allclose(signs, signs_ref)
+
+
+@pytest.mark.parametrize(
+    ("molecule", "basis", "symm", "tol", "wf_ref"),
+    [
+        (
+            [["H", (0, 0, 0)], ["H", (0, 0, 0.71)]],
+            "sto6g",
+            "d2h",
+            1e-1,
+            {(1, 1): 0.9942969785398776, (2, 2): -0.10664669927602176},
+        ),
+        (
+            [["H", (0, 0, 0)], ["H", (0, 0, 0.71)]],
+            "cc-pvdz",
+            "d2h",
+            4e-2,
+            {
+                (1, 1): 0.9919704795977625,
+                (2, 2): -0.048530356564387034,
+                (2, 8): 0.0445233308500785,
+                (4, 4): -0.05003594568491194,
+                (8, 2): 0.04452333085007853,
+                (8, 8): -0.05226230322043741,
+                (16, 16): -0.0404759737476627,
+                (32, 32): -0.0404759737476627,
+            },
+        ),
+    ],
+)
+def test_ucisd_state(molecule, basis, symm, tol, wf_ref):
+    r"""Test that _ucisd_state returns the correct wavefunction."""
+
+    mol = pyscf.gto.M(atom=molecule, basis=basis, symmetry=symm)
+    myhf = pyscf.scf.UHF(mol).run()
+    myci = pyscf.ci.UCISD(myhf).run()
+
+    wf_cisd = qchem.convert._ucisd_state(myci, tol=tol)
+
+    assert wf_cisd.keys() == wf_ref.keys()
+    assert np.allclose(abs(np.array(list(wf_cisd.values()))), abs(np.array(list(wf_ref.values()))))
+
+
+@pytest.mark.parametrize(
+    ("wf_dict", "n_orbitals", "wf_ref"),
+    [
+        (  # -0.99 |1100> + 0.11 |0011>
+            {(1, 1): -0.9942969785398778, (2, 2): 0.10664669927602179},
+            2,
+            np.array(
+                [
+                    0.0,
+                    0.0,
+                    0.0,
+                    0.1066467,
+                    0.0,
+                    0.0,
+                    0.0,
+                    0.0,
+                    0.0,
+                    0.0,
+                    0.0,
+                    0.0,
+                    -0.99429698,
+                    0.0,
+                    0.0,
+                    0.0,
+                ]
+            ),
+        ),
+    ],
+)
+def test_wfdict_to_statevector(wf_dict, n_orbitals, wf_ref):
+    r"""Test that _wfdict_to_statevector returns the correct statevector."""
+    wf_comp = qchem.convert._wfdict_to_statevector(wf_dict, n_orbitals)
+    assert np.allclose(wf_comp, wf_ref)
```

Comment:
might have an overall sign flip issue here

Reply:
Could you elaborate?

Reply:
It could happen that wf_comp is numerically the same as wf_ref, except that wf_comp = - wf_ref. This case is being caught in all the other equivalent tests via an `or` condition, but not in this one.

Reply:
`wf_comp = - wf_ref` happens because of pyscf. This function should be exact in converting the dictionary format to a state vector. Right?

Reply:
Yes my bad, this issue indeed should not arise in this case.